tests/networkpolicy_test.go

```
@@ -223,17 +293,73 @@ func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
+	return policy
+}
+
+func waitForNetworkPolicyDeletion(policy *networkv1.NetworkPolicy) {
 	if policy == nil {
 		return
 	}
 
 	virtClient, err := kubecli.GetKubevirtClient()
 	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &v13.DeleteOptions{})).To(Succeed())
+	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &metav1.DeleteOptions{})).To(Succeed())
 	EventuallyWithOffset(1, func() error {
-		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, v13.GetOptions{})
+		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, metav1.GetOptions{})
 		return err
 	}, 10*time.Second, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())))
 }
+
+func assertHTTPPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance, port int) {
+	ConsistentlyWithOffset(1, checkHTTPPingFn(fromVmi, toVmi, port), 10*time.Second, time.Second).Should(Succeed())
+}
+
+func assertHTTPPingFailed(vmiFrom, vmiTo *v1.VirtualMachineInstance, port int) {
+	EventuallyWithOffset(1, checkHTTPPingFn(vmiFrom, vmiTo, port), 10*time.Second, time.Second).ShouldNot(Succeed())
+	ConsistentlyWithOffset(1, checkHTTPPingFn(vmiFrom, vmiTo, port), 10*time.Second, time.Second).ShouldNot(Succeed())
+}
+
+func checkHTTPPingFn(fromVmi, toVmi *v1.VirtualMachineInstance, port int) func() error {
+	return func() error {
+		const curlCheckCmd = "curl --head %s --connect-timeout 5\n"
+		for _, ip := range toVmi.Status.Interfaces[0].IPs {
+			url := fmt.Sprintf("http://%s", net.JoinHostPort(ip, strconv.Itoa(port)))
+			curlCheck := fmt.Sprintf(curlCheckCmd, url)
+			err := tests.VmiConsoleRunCommand(fromVmi, curlCheck, 10*time.Second)
+			if err != nil {
```

there is a bug here 
if you expect it to fail on ping, you dont check the 2nd ip in the list if one failed, and we do need to check for both

---

Good catch, will implementing with an slice of errors, so we can reuse the function allright, I think we can do the same with assertPing friends.

---

as close as to what assertPing do is better imo, lets keep it as simple and consistency as possible please

---

The `assertPing` is more obscure since they have differnt `err != nil` `err == nil` with the same ping code is very easy to read it wrong, I know that we don't have to ping the next address at some circustances but the optimization is not worth imho, checking an slice of errors is easier to understand.

---

Wait, first of, why do we care about 2 IPs ? Is this meant to be tested on dual stack ?...

---

yes, previously the networkpolicy_test.go was refactored to be dual stack and then we are adding the tier2 tests.

---

@oshoval after doing some checkings looks like we have a curl with timeout 5seconds and that's a lot, I think it's worth breaking the fail fast approach from assertPing, I will rework like that, thanks for comment again.